Improve existing logical operation

NiyongaboEric opened the pull request:
#### What does this PR do? 
The PR helps to refactor the existing work.

#### Description of Task to be completed?
- Refactor sidebar.
- Remove <Del>string comparison<Del>
- Add style

#### How should this be manually tested?
- `git clone https://github.com/andela/team-odd-bn-frontend.git`
- `cd team-odd-bn-backend`
- bg-frontend-design-improvement-171403437 
- `npm install`

> Pass wrong <del>token</del> will redirect to `signin`
> Skip editing profile step will redirect you to `/profile` page
> Skip email verification will redirect to `verify-email` page

#### Any background context you want to provide?
The current work works well. However, we need to improve.

#### What are the relevant pivotal tracker stories?
[#171403437](https://www.pivotaltracker.com/story/show/171403437)

#### Screenshots (if appropriate)
#### Questions:

NiyongaboEric committed:
bg(refactoring): improve existing logical operation checking
- refactor code
- add style
- [Finishes #171403437]

hezronkimutai commented on src/views/Dashboard/sidebar/index.js:
```diff
@@ -41,26 +41,27 @@ export class Sidebar extends Component {
+ 
+    if(tokenValid && profileError && profileError.error) {
+      return <Redirect to="/verify-email"/>;
+    }
```
@NiyoEric , What if you try something like this.
```
tokenValid && profileError && profileError.error && <Redirect to="/verify-email"/>
```

NiyongaboEric replied:
@hezronkimutai, I have tried it out but it blocks `Redirect` functionality.
> Code used
``` jsx
   tokenValid && profileError && profileError.error && <Redirect to="/verify-email"/>;
```
*Output*
<img width="1440" alt="Screen Shot 2020-02-21 at 11 03 49" src="https://user-images.githubusercontent.com/33821734/75019669-fb3c8b80-5499-11ea-9f0e-523ad88b5b1c.png">

hezronkimutai commented on src/views/Dashboard/sidebar/index.js:
```diff
@@ -41,26 +41,27 @@ export class Sidebar extends Component {
+    if(tokenValid && profileError && profileError.message) {
+      localStorage.removeItem('token')
+      return <Redirect to="/signin"/>;
+    }
```
```
tokenValid && profileError && profileError.message && ( (localStorage.removeItem('token')) (<Redirect to="/signin"/>))
```

NiyongaboEric replied:
The same issue. We can not redirect the user in case,
- token is invalid
- token is malformed
- token is expired

``` jsx
tokenValid && profileError && profileError.message && (localStorage.removeItem('token'), <Redirect to="/signin"/>);
```
However, I think code looks complicate than it should.

<img width="1440" alt="Screen Shot 2020-02-21 at 11 18 00" src="https://user-images.githubusercontent.com/33821734/75020682-e06b1680-549b-11ea-8923-756dfe0fac2a.png">

hezronkimutai replied:
Okey, @NiyoEric , we can leave it that way.

hezronkimutai commented on src/views/Dashboard/sidebar/index.js:
```diff
@@ -41,26 +41,27 @@ export class Sidebar extends Component {
+
+    if (profile.data && !profile.data.managerId) {
+      return <Redirect to="/profile"/>;
     }
```
```
profile.data && !profile.data.managerId && <Redirect to="/profile"/>

```

hezronkimutai reviewed (changes requested):
Hi @NiyoEric , I have left some comments, please view them

victkarangwa reviewed (approved):
Well done!

william000000 commented on src/components/Profile.js:
```diff
@@ -53,11 +53,11 @@ export class Profile extends Component {
-    if (this.props.profileError.error
-    || this.props.profileError.message === 'You have provided an invalid token') {
+    if (profileError.error
+    || profileError.message === 'You have provided an invalid token') {
```
Looks good to destructure these!

william000000 commented on src/containers/Trips/MulticityTrip.js:
```diff
@@ -61,10 +61,9 @@ export class MulticityTrip extends Component {
-        return history.push('/requests');
-      }
+    const isTripChanged = prevProps.multicityTripData !== multicityTripData;
+    if (isTripChanged && (multicityTripData.message === 'Trip requested successfully')) {
```
This is more readable than before

kipropbrian commented on src/containers/Trips/MulticityTrip.js:
```diff
@@ -61,10 +61,9 @@ export class MulticityTrip extends Component {
-      if (multicityTripData.message === 'Trip requested successfully') {
-        return history.push('/requests');
-      }
+    const isTripChanged = prevProps.multicityTripData !== multicityTripData;
```
this line is unnecessary. No need to hold this in a variable.

NiyongaboEric replied:
I thought writing in one condition would make the line big. I removed the variable and broke condition in multiple lines.

kipropbrian commented on src/containers/TripsContainer/oneway.js:
```diff
@@ -18,10 +18,9 @@ export class Oneway extends Component {
-      if (incomingData.message === 'Trip requested successfully') {
-        return history.push(`/requests/${incomingData.data.id}`);
-      }
+    const isDataChanged = incomingData !== oldData;
```
Same here.

NiyongaboEric replied:
I 've removed it.

NiyongaboEric committed:
bg(refactoring): remove string comparison

- improve current condition checking
- update env

modify multicity trip request
- remove string comparison for trip request
- check the status of the trip
- edit test`
- [Finishes #171403437]

kipropbrian commented on src/components/CreateProfile.js:
```diff
@@ -51,7 +51,7 @@ export class CreateProfile extends Component {
         await getProfile();
         if (this.props.profileError.error
-        || this.props.profileError.message === 'You have provided an invalid token') {
+        || this.props.profileError.status === 401) {
```
Destructor this

kipropbrian commented on src/containers/TripsContainer/Requests.js:
```diff
@@ -17,11 +17,11 @@ class Requests extends Component {
-    if (this.props.profileError.error
-  || this.props.profileError.message === 'You have provided an invalid token') {
+    if (profileError.error
+  || profileError.message === 'You have provided an invalid token') {
```
Can you do that here too

kipropbrian commented on src/containers/trips/SearchStats.js:
```diff
@@ -12,12 +12,12 @@ export class Requests extends Component {
-    if (this.props.profileError.error
-  || this.props.profileError.message === 'You have provided an invalid token') {
+    if (profileError.error
+  || profileError.message === 'You have provided an invalid token') {
```
Here too

victkarangwa commented on .env.example:
```diff
@@ -1 +1,13 @@
+GOOGLE_CALLBACK_URL=GOOGLE_CALLBACK_URL
+JWT_KEY=JWT_KEY
+TESTING_TOKEN=TESTING_TOKEN
+REACT_APP_MAPBOX_TOKEN=REACT_APP_MAPBOX_TOKEN
```
EOL

victkarangwa commented on src/containers/Trips/MulticityTrip.js:
```diff
@@ -61,10 +61,11 @@ export class MulticityTrip extends Component {
+    if (
+        (prevProps.multicityTripData !== multicityTripData) && 
+        (multicityTripData.message === 'Trip requested successfully')
+      ) {
```
Can we also check this condition by status instead of message?

NiyongaboEric replied:
I forgot to change it.

victkarangwa commented on src/containers/TripsContainer/oneway.js:
```diff
@@ -15,13 +15,14 @@ export class Oneway extends Component {
-      }
+    if (
+      (incomingData !== oldData)
+      && (incomingData.message === 'Trip requested successfully')
+    ) {
```
We can check `incomingData.status` instead of `incomingData.message`

victkarangwa reviewed (commented):
Well done @NiyoEric 
However, I've left some feedback. We can work together to get it done!
